docs/functional/architecture/0009-Compiled-Backend-Integration.md

```diff
@@ -0,0 +1,97 @@
+# Integration of compiled backends
+
+- **Status**: valid 
+- **Authors**: Peter Kardos (@petiaccja)
+- **Created**: 2022-07-13
+- **Updated**: 2022-07-13
+
+Summary of the key design choices made for the live execution of generated C++ (and other compiled) code.
+
+## Context
+
+As per the current state, gt4py can execute iterator IR by generating Python code from it which is loaded back as a module. Additionally, gt4py can also emit equivalent C++ code from iterator IR. The integration of the compiled backends focuses on compiling the emitted C++ code and calling it from Python, effectively executing iterator IR live by translating it to machine code.
+
+The process of executing ITIR this way consists of the following steps:
+1. Generate C++ code equivalent (*fencil code*) to ITIR
```

Comment:
Yeah, this is a bit problematic. I want to refer to the *C++ code equivalent to ITIR* in some way, such as *fencil code*, but it can't be simply just *[generated] C++ code* because the *binding code* is also just *generated C++ code*.